src/desktop_notifier/main.py

```diff
@@ -280,7 +280,10 @@ def on_clicked(self) -> Callable[[str], Any] | None:
         """
         A method to call when a notification is clicked
 
-        The method must take the notification identifier as a single argument.
+        The method must take the notification identifier as a single argument. You must
+        check whether the given identifier matches any of the notifications you care
+        about, because the notifications server might signal events of other
+        applications as well.
```

Comment:
> because the notifications server might signal events of other applications as well.

This really seems like a bug rather than a feature to me. Where is the case currently?

Reply:
Following https://gitlab.gnome.org/GNOME/gnome-shell/-/issues/5008#note_1373251 basically any desktop environment on Linux except latest GNOME versions, the latter IMO rather being a bug (see https://gitlab.gnome.org/GNOME/gnome-shell/-/issues/8072). The FreeDesktop.org API doesn't distinguish between applications. Since Desktop Notifier intentionally calls class-level event handlers when no matching notification is found in its cache, any signal the notifications server sends is passed through; i.e. this isn't new and true for Desktop Notifier 6.0 as well, it just wasn't documented.

Reply:
Hmm, if it's really done this way, this appears like a huge privacy issue. It exposes a lot of information about other running apps to yours, possibly even text entered in reply fields.

Coming from a macOS world, you cannot even send notifications if your process does not belong to a signed app bundle and most certainly do not get events from any other app. 

I suspect the Linux desktop world is slowly moving away from that with flatpak and similar packaging solutions as well as sandboxing.

But as long as this is the case, I am happy to have the doc string. Can you maybe mention that this is limited to some Linux notification servers on the dbus backend? Just so that the comment does not frighten the casual reader.

Reply:
The FreeDesktop.org API neither supports reply fields, nor does it provide an API to access the contents of notifications, so there isn't anything exposed but that an notification (identified by an unint32) of an unknown application and with unknown contents was closed, clicked, or one of its buttons (given here, if a dev chooses to use sensitive information as button identifier, this information would indeed be exposed - quite rare I guess, most applications seem to use an incrementing index; Desktop Notifier isn't affected either due to its UUID button identifiers :heart:) was clicked. So, no privacy issue here at all :relieved: That's typical for Linux btw: If one doesn't provide a way to access sensitive information to begin with, it can't get lost :relaxed: